### ISSUE TITLE
[Feature]: Archiverse implementation

### ISSUE THREAD

**Issue by CCIGAMES** [PretendoNetwork/miiverse-api#43]
### Checked Existing

- [X] I have checked the repository for duplicate issues.

### What feature do you want to see added?

The Implementation of the biggest Miiverse archive on the internet: [Archiverse](https://archiverse.app/) into Pretendo's Miiverse as to fully restore Miiverse to what it was before it was shut down.

### Why do you want to have this feature?

Because it makes Pretendo's Miiverse look more alive/full, and it's just good for it to properly pick up where official miiverse left off.

### Any other details to share? (OPTIONAL)

has a github: https://github.com/kyle1373/archiverse

**Comment by MatthewL246** [comment]
While this would be possible, it doesn't fit with our design goals for Juxtaposition. The team has previously decided that Juxtaposition is designed to focus on the future and reimagine what Miiverse could be, not focus on the nostalgia of what Miiverse was. Adding archived data from the original Miiverse goes too far into the nostalgia direction.

There are also several technical issues that would make this difficult:

- Archiverse is huge, with over 17TB of data! That would be quite expensive to host.
- The data isn't in the same format as Juxtaposition data is stored, and I believe some internal metadata about the posts was never archived.
- Adding NNID data to our servers would be difficult. What if someone has a PNID with the same name as someone else's NNID? We could do something like prefixing NNID usernames with `NNID_`, but that would cause issues with NNIDs that already have the maximum username length.

Because of these reasons, we don't plan on adding Archiverse data to Juxtaposition.

**Comment by Repflez** [comment]
There are several reasons why it's a bad idea to use the data:

* Ownership: The Archiverse data is not property of them, but Nintendo's, as well the creators of the post/comment/drawing/etc. Pretendo doesn't have the rights to use it and may become a copyright nightmare trying to get permission to everyone who posted the content so it can be used.
* Storage: The data for Archiverse is roughly 17 TB, not a lot in the grand scheme of things, but nothing to sneeze at either. Storage has a cost, after all.
* Popularity: Despite being the only alternative to the online services for both consoles, it's not worth it to invest the money required on a feature that may or may not be used as often to justify the costs involved on hosting it.
* Lack of data: To be able to use the data in the dump, it needs to use more data than the provided by the dump (mainly banners for Wii U, Title IDs for jump to work, etc). This requires a lot of work to get them in a usable state.

tl;dr It's not worth it. Use the website that is there.

**Comment by CCIGAMES** [comment]
That's reasonable, but when it comes to hosting, most of Archiverse is stored on github and maybe even a bit of IA for good measure, so I presume you could feasibly make a way to pull the miiverse archived data directly from the github repository, and to minimise loading times, it could be made so it's only loading archived stuff when the user is intending to look for older posts, as for Nintendo ownership, yeah that's understandable but the trouble is that Juxt is a replica of Miiverse, everything you are doing in Juxt is replicating and building off of a service Nintendo owns, so you won't ever escape the Nintendo purgatory so you may as well use the tools the gracious open source gods have given us to make ourselves comfortable.

**Comment by MatthewL246** [comment]
> That's reasonable, but when it comes to hosting, most of Archiverse is stored on github and maybe even a bit of IA for good measure

That is interesting - I didn't realize that Archiverse loads most of its data directly from the Internet Archive. Loading images from IA would reduce the hosting cost, but it would be very difficult to integrate that into the Juxt database because it currently stores the base64-encoded image data itself.

> the trouble is that Juxt is a replica of Miiverse, everything you are doing in Juxt is replicating and building off of a service Nintendo owns

Creating a service that is inspired by a service made by Nintendo is legally different from using data that is owned by Nintendo. Juxt does not use any Nintendo-owned data or assets.

> As for lack of data etc, fair enough although tmk I've seen archiverse be completely fine with the data it has.

The data that's missing is metadata that isn't required to display a post on a website, but it would be required to create a fully-formed post entry in the Juxt database.

**Comment by CCIGAMES** [comment]
> > That's reasonable, but when it comes to hosting, most of Archiverse is stored on github and maybe even a bit of IA for good measure
> 
> That is interesting - I didn't realize that Archiverse loads most of its data directly from the Internet Archive. Loading images from IA would reduce the hosting cost, but it would be very difficult to integrate that into the Juxt database because it currently stores the base64-encoded image data itself.
> 
> > the trouble is that Juxt is a replica of Miiverse, everything you are doing in Juxt is replicating and building off of a service Nintendo owns
> 
> Creating a service that is inspired by a service made by Nintendo is legally different from using data that is owned by Nintendo. Juxt does not use any Nintendo-owned data or assets.
> 
> > As for lack of data etc, fair enough although tmk I've seen archiverse be completely fine with the data it has.
> 
> The data that's missing is metadata that isn't required to display a post on a website, but it would be required to create a fully-formed post entry in the Juxt database.
> 
> Finally, I'd just like to reiterate that this just isn't something that the team feels would fit well with the intended design and goals of Juxtaposition. If it were, we might be able to work around the technical issues - but as it stands, this isn't something we would prefer to put effort into.

I understand, I just thought I'd mention it as a possibility down the line that many people would be happy to see.

Also, just for "I didn't realize that Archiverse loads most of its data directly from the Internet Archive. Loading images from IA would reduce the hosting cost", sorry for the confusion, Archiverse is almost entirely contained within a github repo, and it's own website (which is why I was saying you could pull info from those rather than self host the archive inside pretendo's miiverse servers), but IA has the archive used to create Archiverse iirc as well as backups of the website etc, this is doable, although I completely understand that you arent planning on doing it and it isnt intended atm. just thought id bring it up

**Comment by jonbarrow** [comment]
Reopening this for internal reasons

**Comment by CCIGAMES** [comment]
This is interesting
